include/gsl/gsl-lite.hpp

```diff
@@ -1293,7 +1296,38 @@ class not_null
     typedef T get_result_t;
 #endif
 
+#if gsl_CPP11_OR_GREATER
+    template<class U, class E = void>
+    struct element_type_helper
```

Comment:
Could you perhaps move `element_type_helper<>` out-of-class (preferably to namespace `gsl::detail`)? It doesn't depend directly on the template arguments of `not_null<>`, so keeping it outside potentially reduces the number of specializations the compiler has to generate. Also, I remember VS 2015 having occasional trouble with in-class specializations.

Reply:
ok